app/src/components/submissionPage/SnapList.tsx

```diff
@@ -63,14 +69,29 @@ const SnapList: React.FunctionComponent<SnapListProps> = ({ snaps }) => {
         ));
     }
 
+    const onDeleteSnapPressed = async (snap: Snap) => {
+        if (!confirm("Are you sure you want to delete this snap?")) {
+            return;
+        }
+        try {
+            await deleteSnap(snap);
+        } catch (error) {
+            alert(`There was an error deleting your snap ${error.message}`);
+        }
+    };
+
     const listItems = snaps.map((snap: Snap, index: number) => (
-        <div className="col-sm-6 col-md-4 mb-4" key={index}>
+        <div className="col-sm-6 col-md-4 mb-4" key={snap.id ?? index}>
```

Comment:
I _think_ we'll always want to use the ID here, I think using the index could cause issues with updating.